akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Demux.scala

```diff
@@ -357,21 +361,40 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
         }
       })
 
+      // -----------------------------------------------------------------
       // FIXME: What if user handler doesn't pull in new substreams? Should we reject them
-      //        after a while or buffer only a limited amount? We should also be able to
-      //        keep the buffer limited to the number of concurrent streams as negotiated
-      //        with the other side.
-      val bufferedSubStreamOutput = new BufferedOutlet[Http2SubStream](substreamOut)
+      //        after a while or buffer only a limited amount?
```

Comment:
I guess this as well as limiting the size of the buffer becomes harder/impossible with the custom buffered outlet replaced with emit. The buffer is now in fact an Akka streams internal linked list of emits with no API to look at size/interrupt it etc?

Reply:
I'm not sure I understand your question. Actually, I think Github UI failed to mark it as resolved since I've rolled-back the use of `emit` to restore the use of the `BufferedOutlet`.

The buffer (in `BufferedOutlet` or in Akka Stream internals) can only blow up when:

 - the remote peer sets a very high MAX_CONCURRENT_STREAMS
 - the user handler doesn't grab the emitted elements

The server is protected because we don't support PUSH_PROMISE.

The client could blow up if:

 - the servers supports PUSH_PROMISE and the client set an unlimited MAX_CONCURRENT_STREAMS, or
 - the user creates more requests than it pulls responses (and the server sets an unlimited MAX_CONCURRENT_STREAMS)

There is room for improvement here, but I don't think it's in scope for this PR.

Reply:
Ah, I indeed missed the restore of BufferedOutlet, you can ignore this comment!